src/commands/Minion/trekshop.ts

```diff
@@ -0,0 +1,172 @@
+				amount: 0
+			},
+			{
+				skillName: SkillsEnum.Thieving,
+				amount: 0
+			},
+			{
+				skillName: SkillsEnum.Slayer,
+				amount: 0
+			},
+			{
+				skillName: SkillsEnum.Firemaking,
+				amount: 0
+			},
+			{
+				skillName: SkillsEnum.Fishing,
+				amount: 0
+			},
+			{
+				skillName: SkillsEnum.Woodcutting,
+				amount: 0
+			},
+			{
+				skillName: SkillsEnum.Mining,
+				amount: 0
+			}
+		];
+
+		for (let i = 0; i < quantity; i++) {
+			let outputTotal = 0;
+
+			switch (type) {
+				case 'easy':
+					inItems.addItem(rewardTokens.easy, 1);
+					outputTotal = rand(specifiedItem.easyRange[0], specifiedItem.easyRange[1]);
+					break;
+				case 'medium':
+					inItems.addItem(rewardTokens.medium, 1);
+					outputTotal = rand(specifiedItem.medRange[0], specifiedItem.medRange[1]);
+					break;
+				case 'hard':
+					inItems.addItem(rewardTokens.hard, 1);
+					outputTotal = rand(specifiedItem.hardRange[0], specifiedItem.hardRange[1]);
+					break;
+			}
+
+			if (specifiedItem.name === 'Herbs') {
+				outItems.add(
+					percentChance(50) ? 'Tarromin' : 'Harralander',
+					Math.floor(reduceNumByPercent(outputTotal, 34))
+				);
+				outItems.add('Toadflax', Math.floor(reduceNumByPercent(outputTotal, 66)));
+			} else if (specifiedItem.name === 'Ore') {
+				outItems.add('Coal', Math.floor(reduceNumByPercent(outputTotal, 34)));
+				outItems.add('Iron ore', Math.floor(reduceNumByPercent(outputTotal, 66)));
+			} else if (specifiedItem.name === 'Experience') {
+				(
+					outXP.find(
+						item => item.skillName === TrekExperience[Math.floor(Math.random() * TrekExperience.length)]
+					) || outXP[0]
+				).amount += outputTotal;
+			} else {
+				outItems.add(specifiedItem.name, outputTotal);
+			}
+		}
+
+		if (!bankHasAllItemsFromBank(userBank, inItems.bank)) {
+			return msg.send("You don't have enough reward tokens for that.");
+		}
+
+		if (!msg.flagArgs.cf && !msg.flagArgs.confirm) {
+			const sellMsg = await msg.channel.send(
+				`${user}, say \`confirm\` to confirm that you want to use ${quantity} ${type} reward tokens to buy sets of ${specifiedItem.name}.`
+			);
+
+			// Confirm the user wants to buy
+			try {
+				await msg.channel.awaitMessages(
+					_msg => _msg.author.id === user.id && _msg.content.toLowerCase() === 'confirm',
+					{
+						max: 1,
+						time: Time.Second * 15,
+						errors: ['time']
+					}
+				);
+			} catch (err) {
+				return sellMsg.edit(`Cancelling purchase of ${quantity} sets of ${toTitleCase(specifiedItem.name)}.`);
+			}
+		}
+
+		if (outItems.length > 0) await user.addItemsToBank(outItems);
+		await user.removeItemsFromBank(inItems);
+
+		let ret = `You redeemed **${inItems}** for `;
+		if (outItems.length > 0) {
+			ret += `**${outItems}**`;
+		} else {
+			ret += 'XP. You received: ';
+		}
+
+		ret += (await Promise.all(outXP.map(xp => (xp.amount > 0 ? user.addXP(xp) : ''))))
+			.filter(Boolean)
+			.join(', ')
+			.replace(/You received /g, '');
```

Comment:
1. try `addXP({ minimal: true })` instead of the `.replace`
2. do `outXP.filter(xp => xp.amount > 0).map()` to avoid needing the ternary and filter.